src/headless/plugins/chat/utils.js

```diff
@@ -22,7 +22,16 @@ export function routeToChat (event) {
         return;
     }
     event?.preventDefault();
-    const jid = location.hash.split('=').pop();
+    let jid = location.hash.split('=').pop();
+    // decodeURIComponent is needed in case the JID contains special characters
+    // that were URL-encoded, e.g. `user%40domain` instead of `user@domain`.
+    jid = decodeURIComponent(jid);
+
+    // Remove xmpp: prefix if present
+    if (jid.startsWith('xmpp:')) {
+        jid = jid.slice(5);
```

Comment:
Thanks @marcellintacite, your PR looks better now.

However, there are still various use-cases from XEP-0147 which aren't covered.

For example the message action:
https://xmpp.org/extensions/xep-0147.html#actions-message

And roster management actions:
https://xmpp.org/extensions/xep-0147.html#actions-roster-add

In the very least, you need to remove the query string (the part after `?`) from the xmpp URI, like you removed the `xmpp:` prefix, but ideally you could also implement the actions from XEP-0147.